Solutions/Marain.UserNotifications.Client/Marain/UserNotifications/Client/Management/UserNotificationsManagementClient.cs

```diff
@@ -299,5 +309,16 @@ private HttpRequestMessage AddETagToNotificationTemplateHeaders(ICommunicationTe
 
             return null;
         }
+
+        private ImmutableDictionary<string, string> AddETagToImmutableDictionary(HttpResponseMessage response)
```

Comment:
I think this method has two problems:

- it is misnamed
- it is an instance member but apparently does not need to be

This caused me to misunderstand what was happening at the point where it was called. Given that this is an instance method that claims to add something to a dictionary I assumed that meant there was already some dictionary somewhere associated with this object that this method would return a modified version of.

In fact, if I've understood it, this returns a brand new dictionary containing either a single item, the `ETag`, or containing nothing if there is no `ETag`. So maybe this should be a static method called something like `WrapETagInImmutableDictionary`?

Reply:
Hi Ian, I have renamed this method to the one you have suggested and made it static.